src/User/User.ts

```diff
@@ -0,0 +1,32 @@
+import firebase from "firebase";
+
+type getToken =
+  | ((forceRefresh?: boolean | undefined) => Promise<string>)
+  | undefined;
+
+type IUSER = firebase.User | null;
```

Comment:
This might change later but I don't think we want to allow `null` at this time.

Reply:
This was inferred by the firebase itself otherwise I was getting a linting error.